tool/cmd/kitex/versions/version.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package versions
```

Comment:
Is there any opensource library for this? Seems to be a quite common need.

Reply:
There is a library https://github.com/hashicorp/go-version

Reply:
- This library doesn't recognize the prefix ```v```. 
- The regular expression of the semantic version is deterministic(https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)
so we just simply implement it by ourselves.